### DISCUSSION TITLE
Is there a way to simulate Hue Labs? i.e. flicker of a candle?

### DISCUSSION THREAD

**Discussion by jschwalbe** [Koenkk/zigbee2mqtt#9247] [General]
I have a number of hue bulbs which I'm planning to switch over to z2m. Hue has this thing called "hue labs" which allows one to make different "formulas" to do certain effects. 

For example, https://labs.meethue.com/formulas/huelabs/candlelight-romance causes the selected bulbs to flicker, as if by candlelight. Raising up and down, at seemingly random intervals. When you have a bunch of them, it actually looks pretty cool.

`
This Formula turns your lights into candles! Use the Candlelight romance Formula for a relaxing or romantic mood — especially if you use the Formula with our collection of Filament bulbs, whose glowing inner coils become that much more beautiful as they dim and brighten dynamically.
Your lights will “flicker” at your preferred intensity: small for a subtle effect, large for more of a statement. You can even activate the Formula using your voice via Alexa simply by saying, “Alexa, turn on Candlelight romance.” When you stop the Formula, your lights revert to their previous settings.
`

I tried doing something similar w/ z2m and node red, but it ended up just congesting the system and nothing worked (oops - too many messages!). 

So, I am wondering if there is a way to simulate this sort of thing with hue (or non-hue is the option exists) bulbs and z2m.

Any ideas? Thanks!

**Comment by u20p17** [comment]
Hi, 
did you find a solution in the meantime ? would also be interested into this.

BR

**Reply by EvertDekker** [reply]
Most likely to kind of effects are done by the firmware of the bulb. As you already find out it can't be done by sending message with the different states of the effect dut the overload of the zigbee network.
So have a look at bulbs the support this effect or a firmware update for the bulbs you have.

**Reply by u20p17** [reply]
I am not sure how Hue Labs is working in the background - so i do not know if the different effects get ‚downloaded‘ to the hue bulbs or if the hue bridge is sending the commands to the bulbs… therefore i was asking if there is a solution to use the ‚formulas‘ of hue labs without the huebridge with zigbee2mqtt… it seems that this is not possible which would be a reason to have two seperate zigbee meshes (one for all the hue lights and one with zigbee2mqtt + all other zigbee devices…)

**Comment by deviantintegral** [comment]
This and a "fireplace" effect now show up in the default app, at least when paired over bluetooth. I think this confirms the effects are entirely in firmware, because they continue even when the phone is no longer connected with bluetooth.

I set the Candle effect, and walked through the `LColorCtrl` attributes and nothing obvious changed indicating they were being used for the effect.

I was also surprised that there was no form of "color loop" in the app. Perhaps that requires using a Hue bridge?

**Comment by Berlusconi13** [comment]
Did anybody find a solution?

**Reply by sjorge** [reply]
Some if these are available on the most recent z2m release. They are listed under effect.
